pkg/apis/pipeline/v1alpha1/resource_types.go

```diff
@@ -112,3 +115,12 @@ type PipelineResourceList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []PipelineResource `json:"items"`
 }
+
+// ResourceFromType returns a
+func ResourceFromType(r *PipelineResource) (PipelineResourceInterface, error) {
+	switch r.Spec.Type {
+	case PipelineResourceTypeGit:
+		return NewGitResource(r)
+	}
+	return nil, fmt.Errorf("%s is an invalid PipelineResource", r.Spec.Type)
```

Comment:
hm is this error accurate? we have 2 other kinds of resources

Reply:
as part of `addInputResource` you will find a call to the resource implementation type doing some similar to this. https://github.com/knative/build-pipeline/blob/master/pkg/apis/pipeline/v1alpha1/git_resource.go#L37

Reply:
coo, then my last request would be to change the error message to reflect that, e.g. `%s is an invalid or unimplemented resource` or even something like:

```

	case PipelineResourceTypeGit:
		return NewGitResource(r)
	case PipelineResourceTypeImage:
		return NewImageResource(r)
        case PipelineResourceTypeGCS:
                return nil, fmt.Errorf("PipelineResource type %s is not yet implemented")
	}
...
        return nil, fmt.Errorf("%s is an invalid PipelineResource", r.Spec.Type)
```